src/Video.tsx

```diff
@@ -571,6 +575,34 @@ const Video = forwardRef<VideoRef, ReactVideoProps>(
       ],
     );
 
+    const hasValidDrmProp = drm !== undefined && Object.keys(drm).length !== 0;
+
+    const shallForceViewType =
+      hasValidDrmProp && (viewType === ViewType.TEXTURE || useTextureView);
+
+    if (shallForceViewType) {
+      console.warn(
+        'cannot use DRM on texture view. please set useTextureView={false}',
+      );
+    }
+    if (useSecureView && useTextureView) {
+      console.warn(
+        'cannot use SecureView on texture view. please set useTextureView={false}',
+      );
+    }
+
+    const _viewType = shallForceViewType
```

Comment:
maybe we should move this logic inside a memo :/ 
here in case of bad config a warning will be displayed on every render ...

Reply:
@freeboub I think you can wrap all this code in `useMemo` like this:
```jsx
    const _viewType = useMemo(() => {
      const hasValidDrmProp =
        drm !== undefined && Object.keys(drm).length !== 0;

      const shallForceViewType =
        hasValidDrmProp && (viewType === ViewType.TEXTURE || useTextureView);

      if (shallForceViewType) {
        console.warn(
          'cannot use DRM on texture view. please set useTextureView={false}',
        );
      }
      if (useSecureView && useTextureView) {
        console.warn(
          'cannot use SecureView on texture view. please set useTextureView={false}',
        );
      }

      const _viewType = shallForceViewType
        ? useSecureView
          ? ViewType.SURFACE_SECURE
          : ViewType.SURFACE // check if we should force the type to Surface due to DRM
        : viewType
        ? viewType // else use ViewType from source
        : useSecureView // else infer view type from useSecureView and useTextureView
        ? ViewType.SURFACE_SECURE
        : useTextureView
        ? ViewType.TEXTURE
        : ViewType.SURFACE;
    }, [drm, useTextureView, useSecureView, viewType]);
 ```